integrations/jina/src/jina_haystack/text_embedder.py

```diff
@@ -0,0 +1,108 @@
+# SPDX-FileCopyrightText: 2023-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+import os
+from typing import Any, Dict, List, Optional
+
+import requests
+from haystack import component, default_to_dict
+
+JINA_API_URL: str = "https://api.jina.ai/v1/embeddings"
+
+
+@component
+class JinaTextEmbedder:
+    """
+    A component for embedding strings using Jina models.
+
+    Usage example:
+    ```python
+    from haystack.components.embedders import JinaTextEmbedder
+
+    text_to_embed = "I love pizza!"
+
+    text_embedder = JinaTextEmbedder()
+
+    print(text_embedder.run(text_to_embed))
+
+    # {'embedding': [0.017020374536514282, -0.023255806416273117, ...],
+    # 'metadata': {'model': 'jina-embeddings-v2-base-en',
+    #              'usage': {'prompt_tokens': 4, 'total_tokens': 4}}}
+    ```
+    """
+
+    def __init__(
+        self,
+        api_key: Optional[str] = None,
+        model_name: str = "jina-embeddings-v2-base-en",
+        prefix: str = "",
+        suffix: str = "",
+    ):
+        """
+        Create an JinaTextEmbedder component.
+
+        :param api_key: The Jina API key. It can be explicitly provided or automatically read from the
+            environment variable JINA_API_KEY (recommended).
+        :param model_name: The name of the Jina model to use.
```

Comment:
Same as above. Please provide a link to the docs if available...

Reply:
Done